Docker registry chart

sbezverk opened the pull request:
Private docker registry chart to use within a kubernetes cluster. It uses
node port to allow ALL workers to access registry via 127.0.0.1:30XXX address.

sbezverk committed:
Docker registry chart

Private docker registry chart to use within a kubernetes cluster. It uses
node port to allow ALL workers to access registry via 127.0.0.1:30XXX address.

sbezverk committed:
Docker registry chart

Private docker registry chart to use within a kubernetes cluster. It uses
node port to allow ALL workers to access registry via 127.0.0.1:30XXX address.

unguiculus commented on incubator/docker-registry/values.yaml:
```diff
@@ -0,0 +1,12 @@
+svc_name: docker-registry
```
Use camel case.

sbezverk replied:
Done

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,79 @@
+{{- $registryImage := .Values.registry_image }}
+{{- $registryImageV := .Values.registry_image_version }}
+{{- $tarballURL := .Values.tarball_url }}
+---
```
Any reason why you use variables for everything? That's kind of uncommon.

sbezverk replied:
This service requires great flexibility at least t=fir the use case we use it in koala-kubernetes project. On the other hand we do not want to impose complexity on other users, thus providing defaults values. It is required to support both and other use cases.

unguiculus commented on incubator/docker-registry/Chart.yaml:
```diff
@@ -0,0 +1,9 @@
+version: 0.1.0
```
Add appVersion

sbezverk replied:
done

unguiculus replied:
`appVersion` should reflect what it means as opposed to the chart version. In your case it should be `2`.

unguiculus replied:
Is there an `icon` you could add?

sbezverk replied:
no icon at this moment..

unguiculus commented on incubator/docker-registry/Chart.yaml:
```diff
@@ -0,0 +1,9 @@
+  - docker
+  - registry
+maintainers:
+  - name: Serguei Bezverkhi 
```
Please use Github username.

sbezverk replied:
done

unguiculus reviewed (changes requested):
Thanks for the PR. Please apply best practices regarding labels and values.

https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

Look at examples for standard labels:
https://github.com/kubernetes/helm/tree/master/docs/examples/nginx

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,89 @@
+      volumes:
+      - name: image-store
+        emptyDir: {}
+---
```
Use one file per resource.

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,89 @@
+{{- $replicas := .Values.replicas }}
```
I guess you misunderstood me. It's ok to have the stuff configurable. I just don't see the need to create a local variable here for every value.

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,89 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ $svcName }}
```
name should be the full name. Please use the standard `_helpers.tpl`.

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,89 @@
+  template:
+    metadata:
+      labels:
+        service: {{ $svcName }}
```
`app` and `release` label should go here. Remove `service` label.

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,89 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ $svcName }}
```
name should be the full name. Please use the standard `_helpers.tpl`.

unguiculus commented on incubator/docker-registry/templates/registry.yaml:
```diff
@@ -0,0 +1,89 @@
+    release: {{ .Release.Name }}
+spec:
+  selector:
+    service: {{ $svcName }}
```
`app` and `release` label should go here. Remove `service` label.

unguiculus reviewed (changes requested):
Please have a look at best practices.
https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

You may also check the example charts for reference:
https://github.com/kubernetes/helm/tree/master/docs/examples

sbezverk committed:
Docker registry chart

Private docker registry chart to use within a kubernetes cluster. It uses
node port to allow ALL workers to access registry via 127.0.0.1:30XXX address.

unguiculus commented on incubator/docker-registry/templates/deployment.yaml:
```diff
@@ -0,0 +1,57 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Values.svcName }}
```
```yaml
name: {{ template "fullname" . }}
```

sbezverk replied:
done

unguiculus commented on incubator/docker-registry/templates/service.yaml:
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.svcName }}
```
```yaml
name: {{ template "fullname" . }}
```

sbezverk replied:
done

unguiculus commented on incubator/docker-registry/values.yaml:
```diff
@@ -0,0 +1,12 @@
+svcName: docker-registry
```
`svcName` not needed with above changes.

sbezverk replied:
done

unguiculus commented on incubator/docker-registry/templates/deployment.yaml:
```diff
@@ -0,0 +1,57 @@
+          name: registry
+          protocol: TCP
+      volumes:
+      - name: image-store
```
How about adding persistence options? Here's an example how this can be done:

https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/server-deployment.yaml#L82-L88
https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/server-pvc.yaml
https://github.com/kubernetes/charts/blob/master/stable/prometheus/values.yaml#L332-L371

sbezverk replied:
done

unguiculus commented on incubator/docker-registry/README.md:
````diff
@@ -0,0 +1,42 @@
+
+To push local docker container to the registry:
+```bash
+$ docker tag {local docker container ID} 127.0.0.1:{node_port}/{local docker image name}
````
`127.0.0.1`?

sbezverk replied:
The reason I use 127.0.0.1 is to provide access to the registry to every worker in the cluster in "insecure" way. By default docker allows "insecure" access only to host local address, accessing it by service ip would require adding "secured" access and what is worse is reconfiguration docker daemon on each host.

unguiculus commented on incubator/docker-registry/templates/pvc.yaml:
```diff
@@ -0,0 +1,28 @@
+{{ toYaml .Values.persistentVolume.annotations | indent 4 }}
+  {{- end }}
+  labels:
+    app: {{ template "fullname" . }}
```
`app: {{ template "name" . }}`

sbezverk replied:
done

unguiculus replied:
Label still not correct. Should be `app: {{ template "name" . }}`.

unguiculus commented on incubator/docker-registry/templates/deployment.yaml:
```diff
@@ -0,0 +1,62 @@
+      - name: image-store
+      {{- if .Values.persistentVolumeEnabled }}
+        persistentVolumeClaim:
+          claimName: {{ if .Values.persistentVolume.existingClaim }}{{ .Values.persistentVolume.existingClaim }}{{- else }}{{ template "name" . }}{{- end }}
```
`{{ template "fullname" . }}`

sbezverk replied:
done

unguiculus commented on incubator/docker-registry/templates/pvc.yaml:
```diff
@@ -0,0 +1,28 @@
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ template "name" . }}
```
Should be `name: {{ template "fullname" . }}`.

unguiculus commented:
/ok-to-test
